onadata/libs/models/sorting.py

```diff
@@ -21,11 +22,16 @@ def sort_from_mongo_sort_str(sort_str):
     return sort_values
 
 
-def json_order_by(sort_list):
+def json_order_by(
+        sort_list, none_json_fields: Dict = {}, model_name: str = ""):
     _list = []
 
     for field in sort_list:
-        _str = u" json->>%s"
+        field_key = field.lstrip('-')
+        _str = u" json->>%s"\
+            if field_key not in none_json_fields.keys() else\
+            f'"{model_name}"."{none_json_fields.get(field_key)}"'
+
         if field.startswith('-'):
```

Comment:
Should we add or update the unit tests for the `json_order_by()` function?

Reply:
Added in the latest commits